python/pyarrow/filesystem.py

```diff
@@ -371,9 +371,81 @@ def walk(self, path, refresh=False):
                 yield tup
 
 
-def _sanitize_s3(path):
+class ADLFSWrapper(DaskFileSystem):
```

Comment:
nit: I think I would prefer to spell this out, (At lease Azure)

Reply:
Changed to AzureDatalakeFilesystemWrapper.  This will require a change to the [dask-adlfs](https://github.com/hayesgb/dask-adlfs/blob/master/dask_adlfs/core.py), which I will create and submit once this is complete.